osbs/api.py

```diff
@@ -589,6 +589,18 @@ def get_image_stream_tag(self, tag_id):
         return self.os.get_image_stream_tag(tag_id)
 
     @osbsapi
+    def put_image_stream_tag(self, tag_id, tag):
```

Comment:
Would `update_image_stream_tag()` be a better name? The caller is only expected to use this to update an existing tag I think (`ensure_*` handles creating new ImageStreamTags).

Could have a docstring explaining how the parameters are used.

Is `tag` intended to be a JSON string or an object? I feel like it ought to be an object, as this core function's implementation is the only place that JSON is mentioned.